### DISCUSSION TITLE
ptz control setup question

### DISCUSSION THREAD

**Discussion by Sceptersax** [blakeblackshear/frigate#9738] [Ask A Question] [answered]
Was hoping for a bit more clarity on the ptz control UI setup. 

1. host, should be the camera ip address correct?
2. port is configurable? or do i have to change the camera ptz control port in the camera to 8000?
3. are "user" and "password" supposed to be the camera username and password? Or, is it for the specific username and password of the PTZ function or onvif? btw, my camera doesn't have an option to setup a separate user and pass for this, i just wanted to make sure that this isn't the case.

question 3 is asked because when i type in my numeric password of the camera's auth, i see a yellow squiggly line under the numeric value.

**Comment by NickM-27** [comment]
1. Yes
2. It is a config option, set it to whatever the onvif port is for your camera
3. Some cameras have separate options, others it is the same.

**Reply by Sceptersax** [reply]
Can i put a numeric value in the password field? Should i ignore the squiggly yellow line?

**Reply by NickM-27** [reply]
It should have quotes around it even if it is numbers

**Reply by aspillers2** [reply]
Adding quotes around both my user and password in the yml file fixed the issue with getting the PTZ controls to work for the Vikylin PTZ-2804X-I2.  I almost wonder if this was the issue in the first place.  Either way I'm glad to have this working now.  The supported camera table can probably be updated to say this camera supports PTZ controls but not Autotracking.

**Comment by Sceptersax** [comment]
Also, my camera has a separate area that has dropdowns for a few additional options. 

One of the dropdowns gives options for:

PELCO_D
PELCO_C
MOLYN_P
SANTA_P

**Comment by Sceptersax** [comment, marked as answer]
so for anyone else who might be wrestling with PTZ controls, my issue was being confused by all of the port options the camera i have offers:

![ptz](https://github.com/blakeblackshear/frigate/assets/11879372/961a0c03-37af-4894-bfaf-fb40ad648d25)


I was too focused on the ptz control protocol port since it was close to the default port number 8000. What i should have done was focused on the ONVIF port which was port 80. It is now working correctly.

**Reply by alexplatform** [reply]
@Sceptersax seems like I have the same camera. do you mind sharing your frigate config for this? at the very least the rstp url path ;)
